compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1910,11 +1910,15 @@ fn check_invalid_crate_level_attr(tcx: TyCtxt<'_>, attrs: &[Attribute]) {
         sym::automatically_derived,
         sym::start,
         sym::rustc_main,
+        sym::derive,
     ];
 
     for attr in attrs {
         for attr_to_check in ATTRS_TO_CHECK {
             if attr.has_name(*attr_to_check) {
+                let lo = attr.span.lo() + BytePos(1);
+                let hi = lo + BytePos(1);
```

Comment:
You can try using `span_to_snippet` to get the attribute text, then try replacing the `#![` prefix in that text with `#[` for the suggestion text, and only emit the suggestion if both operations succeed.

Reply:
Okay I'll have a look at doing that tonight - the method `span_suggestion_verbose` does the replacement on the span you give it for you so I suppose I can pass in the `#![` prefix and give it a `#[` to replace it with which should also put the label on the correct span too

Reply:
though thinking about it - I need to be able to get the prefix as a span to be able to pass it to `span_suggestion_verbose` system rather than as a `String` as it's signature is `span_suggestion_verbose(span_to_replace: Span, msg: &str, replacement: String, applicability: Applicability)` so if I use `span_to_snippet` I won't get the section of the span I want to replace but a String instead which I then can't pass to error reporting to do the replacement